Fix default config import

f-PLT opened the pull request:
- Rolled back cron command
- Changed how `default_configuration `is imported

I tested with and without `CANARIE_API_CONFIG_FN ` and everything worked as expected on my end.

Also saw that nginx's healthcheck always came back as unhealthy when using it in GIN, even though everything was working fine.

Turns out, the new version doesn't install curl, compared to inhouse version 0.4.0, so I added it back, but is it something we wish to keep?

francisPLT committed:
rolls back cron command

francisPLT committed:
changes how default_config is imported

francisPLT committed:
Adds curl package to image build

Adding curl to image enables it's use for healthchecks

dbyrns commented on canarieapi/app_object.py:
```diff
@@ -29,9 +24,10 @@
 APP.wsgi_app = ReverseProxied(APP.wsgi_app)
 
 # Config
-APP.config.from_object(default_configuration)
```
In fact moving that inside the else is a breaking change : 
The parameter here : https://github.com/Ouranosinc/CanarieAPI/blob/master/canarieapi/default_configuration.py#L12
is not overridden by : https://github.com/bird-house/birdhouse-deploy/blob/5c5813677951247890f32ea07366c4ba75b61f5f/birdhouse/config/canarie-api/docker_configuration.py.template
So the default value was expected to be used. 
Sorry to have missed that before :(

f-PLT replied:
So we have to keep this line (32) where it is, which means we can't import default_configuration conditionally.

If I followed along correctly, we could then still use `from canarieapi import default_configuration` because we have `CANARIE_API_CONFIG_FN` or should I roll back to the modified cron command (path.insert) with `import default_configuration` ?

dbyrns replied:
You are right in both case.
We cannot import default_configuration conditionally.
And we should still be able to use the from canarieapi import default_configuration because it's ok to use the package installed one as we can override it by the CANARIE_API_CONFIG_FN.
In short, just move the line 31 and 32 above the if and we should be good!

fmigneault replied:
Not breaking: 
https://github.com/Ouranosinc/CanarieAPI/blob/0f28c609aabca3e79a0721e6b3215377148d139d/canarieapi/api.py#L128

fmigneault replied:
But true that it won't hurt to have the default loaded and then overridden by whatever (even partial) definition there was in `CANARIE_API_CONFIG_FN` file.

f-PLT replied:
Changes have been applied

dbyrns commented:
I see no harm to have curl installed in that image, the health command could then also be used in other projects as well.

fmigneault reviewed (changes requested):
waiting for changes as per discussion

francisPLT committed:
changes import and use of 'default_configuration'

Following discussions, it was deemed a better approach
to first import and initialize the configuration from the
'default_configuration' file, which can then be ovewritted
by with the use of'CANARIE_API_CONFIG_FN' environment
variable

dbyrns reviewed (approved):
Good! Thank you
